src/scripts/main.js

```diff
@@ -1 +1,26 @@
 'use strict';
+
+document.addEventListener('DOMContentLoaded', () => {
+  const gallery = document.querySelector('.gallery');
+  const mainImage = document.querySelector('.main-image img');
+
+  // eslint-disable-next-line no-shadow
+  gallery.addEventListener('click', (event) => {
+    event.preventDefault(); // Забороняємо стандартну поведінку посилань
+
+    // Знаходимо ціль кліку
+    const target = event.target;
+
+    if (target.tagName === 'IMG') {
+      // Якщо клік по зображенню
+      const newImageSrc = target.parentElement.getAttribute('href');
```

Comment:
Ensure that the parent element of the image is indeed an anchor tag (`<a>`) before calling `getAttribute('href')`. This will prevent potential errors if the structure changes or if there are other elements inside the gallery.